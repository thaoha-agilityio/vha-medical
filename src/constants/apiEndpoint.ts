export const API_ENDPOINT = {
  AUTH: '/auth/local',
  APPOINTMENTS: '/appointments',
  NOTIFICATIONS: '/notifications',
  SPECIALTIES: '/specialties',
  USERS: '/users',
  UPLOAD: '/upload',
  CHEMISTS: '/chemists',
  PERMISSIONS: '/users-permissions',
};

export const API_URL = `${process.env.NEXT_PUBLIC_API_URL}/api`;
export const IMGBB_URL = `${process.env.NEXT_PUBLIC_IMGBB_URL}?key=${process.env.NEXT_PUBLIC_IMGBB_API_KEY}`;

export const AUTH_TOKEN = process.env.NEXT_PUBLIC_AUTH_TOKEN;
