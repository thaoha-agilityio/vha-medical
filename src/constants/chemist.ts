import { UserModel } from '@/types';

export const DEFAULT_CHEMIST_DATA: UserModel = {
  email: '',
  username: '',
  password: '',
  description: '',
  rating: 0,
  tasks: 0,
  reviews: 0,
  specialtyId: undefined,
  avatar: undefined,
};

export const CHEMISTS_SEARCH_PARAMS = ['specialtyId'] as const;
