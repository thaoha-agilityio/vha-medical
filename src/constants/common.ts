export const RESULT_NOT_FOUND = 'Result Not Found';
