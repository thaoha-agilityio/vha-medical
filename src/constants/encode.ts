export const ENCODE_COOKIE = {
  algorithm: 'aes-256-cbc',
  secret: process.env.COOKIE_SECRET,
};
