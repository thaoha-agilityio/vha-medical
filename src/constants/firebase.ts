export const REGISTRATION_TOKENS = 'token';
