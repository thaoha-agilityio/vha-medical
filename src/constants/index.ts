export * from './apiEndpoint';
export * from './appointment';
export * from './common';
export * from './firebase';
export * from './image';
export * from './message';
export * from './navigation';
export * from './pagination';
export * from './regex';
export * from './router';
export * from './timing';
export * from './chemist';
export * from './encode';
