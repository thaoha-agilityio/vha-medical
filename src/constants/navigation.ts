import {
  AppointmentIcon,
  ChemistIcon,
  DashboardIcon,
  DoctorIcon,
  HospitalIcon,
  MedicineIcon,
  MessageIcon,
  SettingIcon,
} from '@/icons';

// Router
import { PRIVATE_ROUTES } from './router';

export const NAVBAR_LINKS = [
  {
    name: 'Dashboard',
    href: PRIVATE_ROUTES.DASHBOARD,
    icon: DashboardIcon,
    isEnable: true,
  },
  {
    name: 'Appointments',
    href: PRIVATE_ROUTES.APPOINTMENTS,
    icon: AppointmentIcon,
    isEnable: true,
  },
  {
    name: 'Chemists',
    href: PRIVATE_ROUTES.CHEMISTS,
    icon: ChemistIcon,
    isEnable: true,
  },
  { name: 'Hospitals', href: '#', icon: HospitalIcon },
  { name: 'Doctors', href: '#', icon: DoctorIcon },
  { name: 'Medicines', href: '#', icon: MedicineIcon },
];

export const SUPPORT_LINKS = [
  { name: 'Messages', href: '#', icon: MessageIcon },
  {
    name: 'Settings',
    href: '#',
    icon: SettingIcon,
  },
];
