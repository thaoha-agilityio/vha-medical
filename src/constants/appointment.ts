import { AppointmentModel, AppointmentStatusOption } from '@/types';
import { DEFAULT_CHEMIST_DATA } from './chemist';

export const APPOINTMENT_STATUS_OPTIONS: AppointmentStatusOption[] = [
  {
    key: 'new',
    label: 'New',
    value: 0,
  },
  {
    key: 'meeting',
    label: 'Meeting',
    value: 1,
  },
  {
    key: 'cancelled',
    label: 'Cancelled',
    value: 2,
  },
];

export const APPOINTMENT_STATUS = [
  {
    key: '0',
    label: 'New',
  },
  {
    key: '1',
    label: 'Meeting',
  },
  {
    key: '2',
    label: 'Canceled',
  },
];

export const DEFAULT_APPOINTMENT_DATA: AppointmentModel = {
  startTime: '',
  durationTime: '',
  status: 0,
  receiverId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
  senderId: {
    data: {
      id: '',
      attributes: DEFAULT_CHEMIST_DATA,
    },
  },
};

export const APPOINTMENT_SEARCH_PARAMS = ['receiverId', 'senderId'] as const;
