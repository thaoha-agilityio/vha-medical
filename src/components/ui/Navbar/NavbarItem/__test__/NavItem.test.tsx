import { render, screen } from '@testing-library/react';
import { usePathname } from 'next/navigation';

// Components
import { NavItem } from '..';

// Constants
import { NAVBAR_LINKS } from '@/constants';

const mockProps = {
  name: NAVBAR_LINKS[0].name,
  href: NAVBAR_LINKS[0].href,
  icon: NAVBAR_LINKS[0].icon,
};

describe('NavItem Component', () => {
  const setup = () =>
    render(<NavItem {...mockProps} isEnable isExpandSidebar />);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render the NavItem with correct name and icon', () => {
    const { container, getByText } = setup();
    expect(getByText('Dashboard')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should apply active class when the path matches', () => {
    (usePathname as jest.Mock).mockReturnValue(NAVBAR_LINKS[0].href);

    setup();

    const link = screen.getByRole('link');
    expect(link).toHaveClass(
      'bg-linear-sidebar text-primary-100 font-semibold',
    );
  });

  it('should disable the link when isEnable is false', () => {
    (usePathname as jest.Mock).mockReturnValue(NAVBAR_LINKS[3].href);

    render(<NavItem {...mockProps} isExpandSidebar />);

    const link = screen.getByRole('link');
    expect(link).toHaveClass('pointer-events-none');
  });
});
