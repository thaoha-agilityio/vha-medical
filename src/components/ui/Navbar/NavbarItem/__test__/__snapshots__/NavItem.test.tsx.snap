// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem Component should render the NavItem with correct name and icon 1`] = `
<div>
  <li
    class="cursor-not-allowed"
  >
    <a
      aria-label="Dashboard"
      class="text-primary-100 text-sm flex h-[60px] gap-5 mb-[1px] py-5 px-8 lg:px-10 hover:bg-linear-sidebar"
      href="/dashboard"
    >
      <div
        class="w-[17px] h-[17px]"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 18 18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.44204 2.27356H2.46606V7.24953H7.44204V2.27356Z"
            fill-opacity="0.1"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.59942"
          />
          <path
            d="M15.2616 2.27356H10.2856V7.24953H15.2616V2.27356Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.59942"
          />
          <path
            d="M15.2616 10.0928H10.2856V15.0688H15.2616V10.0928Z"
            fill-opacity="0.1"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.59942"
          />
          <path
            d="M7.44204 10.0928H2.46606V15.0688H7.44204V10.0928Z"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.59942"
          />
        </svg>
      </div>
      <p>
        Dashboard
      </p>
    </a>
  </li>
</div>
`;
